### DISCUSSION TITLE
Remove Perl dependency.

### DISCUSSION THREAD

**Discussion by hyoklee** [HDFGroup/hdf5#2896] [Ideas]
Spack installation of Perl on Mac is problematic:

https://github.com/hyoklee/hdf5/runs/6042719583?check_suite_focus=true

less dependency = more portability

**Comment by derobins** [comment]
As much as I'd be thrilled to remove Perl code from the library, it's used to build from non-release branches.

**Comment by hyoklee** [comment]
Yeah, Perl makes daily performance testing of the latest code commit using Spack on Mac impossible.

**Comment by qkoziol** [comment]
Perl ships with MacOS - why do you need to install it from Spack?

**Comment by derobins** [comment]
Instead of running the Perl scripts at configure time, we could set up a CI action that runs them and errors if there were changes. Then you'd only need Perl if you modified a public API call, updated the error categories, etc.

**Comment by derobins** [comment]
We now check in the Perl-generated files, so Perl is no longer necessary to build the library (as long as you don't change the input files).
